### ISSUE TITLE
training on Apolloscape problem

### ISSUE THREAD

**Issue by sotoy** [JiawangBian/SC-SfMLearner-Release#52]
First of all, I want to congratulate you for your excellent work! 
Lately, I have been trying a lot to make your model work on the Apolloscape dataset (particularly road01_ins dataset). With all the hyperparameters left as you propose, it seems that it cannot learn to infer quality disparity maps. Finetuning your already pretrained disparity and pose networks also seems to not work. Here is an example image:
![170927_064253341_Camera_5](https://user-images.githubusercontent.com/10107197/85779317-2545d900-b72c-11ea-8879-ba22105987bf.jpg)

The pretrained disparity network yields this result:
![-170927_064253341_Camera_5_awma_disp](https://user-images.githubusercontent.com/10107197/85779533-5a522b80-b72c-11ea-8d73-a4e4c15f8d77.png)

And after finetuning for 10 epochs:
![-170927_064253341_Camera_5_disp_pre](https://user-images.githubusercontent.com/10107197/85779676-853c7f80-b72c-11ea-8c61-46b72d76a75e.png)

Reducing the learning rate does not seem to make things better either. Is there something obvious I am missing? Are there any preprocessing steps that are mandatory? I am not doing any preprocessing except for reshape.

**Comment by JiawangBian** [comment]
Thanks for your interests in this project.
Of course, preprocessing is important. I suggest that you make sure that
1.  The camera motion between every adjacent image pair would not be too small (static) or large (cannot match). For this purpose, you need consider the FPS of video and delete some redudant static images.
2.  As you can find in your attached sample image that the car itself appears in the image plane, which is indeed the noise for training because of viloating the static world assumption. So I suggest that you may delete certain bottom rows for removing that. 
3. Large sky aera is also noise. So I guess that you may also need delete some top rows.

Remember change the camera intrinsic when you change the image. See 'custom_tranform.py' so that you know how do you change instrinsic when you resize or crop images.

**Comment by sotoy** [comment]
Thanks a lot for the quick response! I did as you said and it seems to be working! However, I think that in custom_transforms.py, in the point where you adjust the intrinsics matrix after cropping, the code instead of 
```
output_intrinsics[0, 2] -= offset_x  
output_intrinsics[1, 2] -= offset_y
```
should be: 
```
intrinsics[0, 2] *= in_w/scaled_w  
intrinsics[1, 2] *= in_h/scaled_h
```
Please correct me if I'm wrong.

**Comment by JiawangBian** [comment]
if you resize images (downsample or upsample), you should do as:
new_fx = factor_x * fx, new_cx = factor_x * cx.  Also for y.

If you only crop images. fx and fy should not be changed, and
new_cx = cx - offset_x. and also do it for cy.

In custom_tranforms.py, it upsamples images first, and then crops images.
So use them above  sequentially.
